roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala

```diff
@@ -44,6 +44,69 @@ object rblstring {
   object stringCiGtr  extends strBinOp("string-ci>", (_.toLowerCase > _.toLowerCase))
   object stringCiGEq  extends strBinOp("string-ci>=", (_.toLowerCase >= _.toLowerCase))
 
+  /** Manipulation */
+  /**
+    * Define the string-concat primitive.
+    * This concatenates n Strings
+    * e.g. (string-concat "abc" "de" "fghi") ==> "abcdefghi"
+    */
+  object stringConcat extends Prim {
+    override val name: String = "string-concat"
+    override val minArgs: Int = 0
+    override val maxArgs: Int = MaxArgs
+
+    @checkTypeMismatch[RblString] // All args must be strings
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RblString] = {
+      val elem = ctxt.argvec.elem
+      val n    = ctxt.nargs
+      val init = ""
+
+      Right(
+        RblString(
+          elem.foldLeft(init)((acc: String, el: Ob) => (acc ++ el.asInstanceOf[RblString].value)))
+      )
+    }
+  }
+
+  /**
+    * Define the string-join primitive.
+    * This concatenates n Strings from a Tuple with a separator. The placement of the seperator is controlled
+    * by a numeric code (actually a bit-field. 0x1 == separator before, 0x2 == separator after).
+    *
+    * examples:
+    *  (string-join 0 ":" ["123" "456" "789"]) ===> "123:456:789"
+    *  (string-join 1 ":" ["123" "456" "789"]) ===> ":123:456:789"
+    *  (string-join 2 ":" ["123" "456" "789"]) ===> "123:456:789:"
+    *  (string-join 3 ":" ["123" "456" "789"]) ===> ":123:456:789:"
+    *  (string-join 3 ":" []) ===> ""
+    */
+  object stringJoin extends Prim {
+    override val name: String = "string-join"
+    override val minArgs: Int = 3
+    override val maxArgs: Int = 3
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RblString] = {
+      val elem = ctxt.argvec.elem
+
+      checkFixnum(0, elem).flatMap( // Ensure arg0 is a Fixnum
```

Comment:
Can you please use a for-expression here?
```
for {
  code <- checkFixnum(0, elem)
  sep <- checkRblString(1, elem)
  ...
} yield {
  val strings = ...
}
````

Reply:
I like that.  I wish you'd taught me that earlier.  ;-)